### DISCUSSION TITLE
Pix4D ortophotomosaic segmentation returns: No objects found in the image.

### DISCUSSION THREAD

**Discussion by oscarbau** [opengeos/segment-geospatial#120] [Q&A] [answered]
Hi all,

I have been testing geoSAM to segment raw drone imagery, which works well. I tested the segmentation of cars and houses with overall good results.

![download](https://github.com/opengeos/segment-geospatial/assets/41731874/1a341948-aa8b-4701-afc1-c8948583e93e)

Now, I am testing the same segmentation on a processed orthophoto mosaic [(see here)](https://wfp-my.sharepoint.com/:i:/g/personal/oscar_bautista_wfp_org/EUlb_lNdD6ROvy_pUr7nuRoBeIKfGLmHrmy0fCSBUu6Bbg?e=pKaM7b).

![image](https://github.com/opengeos/segment-geospatial/assets/41731874/534d10f3-1f66-4a16-93f1-9fda8be07879)

I was able to segment some objects using LanfSAM and the prompt "house" on this dataset.  However, when use "car", I am getting a message saying that there was no object found on the image.

This is part of the mosaic with the settings used to do the text prompt segmentation:

![image](https://github.com/opengeos/segment-geospatial/assets/41731874/89dc1567-3082-4035-9c59-2de51f612eb5)

I have tested multiple thresholds with no success.

I am wondering if this issue might be caused by the resampling method used to create the mosaic as the spatial resolution of the image was reduced, the original image was 2cm approx, and the mosaic is 10 cm. But then why houses are being segmented?

Any ideas to solve this issue?

**Comment by giswqs** [comment, marked as answer]
Try adjusting the thresholds.

**Reply by oscarbau** [reply]
Thank you Qiusheng, indeed lowering the thresholds has an effect on the detection, however, with a threshold of 0.1 on both box and text still the objects identified are few and already detecting a warehouse as a car. see below:

![image](https://github.com/opengeos/segment-geospatial/assets/41731874/0899463f-12bf-4693-a221-114999a88960)

![image](https://github.com/opengeos/segment-geospatial/assets/41731874/cf58346c-22ee-4841-a609-a063cd29d281)

Do you have any suggestions to efficiently select more suitable thresholds? As every segmentation task takes 1 hour + a way to select optimal thresholds would be great.
